### PR TITLE
[MOD-11296] Implement Validation on Rust full II iterator

### PR DESCRIPTION
**Pull request by gdesmott**
Implement revalidate on II iterator.

Blocked by https://github.com/RediSearch/RediSearch/pull/6796

#### Mark if applicable

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

---

> [!NOTE]
> Implements revalidation for full inverted index iterators and adds tests covering EOF, GC/movement, and deletions.
> 
> - **Iterators (in `rqe_iterators/src/inverted_index.rs`)**:
>   - Add `revalidate()` to `FullIterator` that restores position after GC via `skip_to`, returning `RQEValidateStatus::{Ok, Moved}`.
>   - Expose `revalidate()` through `NumericFull` and `TermFull`.
> - **Tests (in `rqe_iterators/tests/inverted_index.rs`)**:
>   - Introduce `RevalidateTest` harness (uses `UnsafeCell`) to mutate the index during iteration.
>   - Add tests for `revalidate` basic behavior, at EOF, after index disappearance (full iterators remain `Ok`), and after document deletion (returns `Moved` and advances).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 19e3a719faf862b3e4250b8f6fc33284d1b775c7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by gdesmott** [comment]
@chesedo @GlenDC : here is the revalidate implementation of the II iterator.

~CI won't pass because of blocking PRs but it can already be reviewed.~

I'm by-passing the "multiple refs on the II" problem (https://github.com/RediSearch/RediSearch/pull/6878#pullrequestreview-3310287429) in a very hacky way. It's not great but the hack is just on a test so I think it's an ok workaround for now so this branch is not blocked forever. Figuring out the proper way to manage concurrent accesses to the II is likely gonna take a while.

File: src/redisearch_rs/inverted_index/src/index_result.rs [hunk 1]
```diff
@@ -173,6 +173,12 @@ pub enum RSTermRecord<'index> {
     },
 }
 
+impl PartialEq for RSTermRecord<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        self.query_term() == other.query_term() && self.offsets() == other.offsets()
+    }
+}
```
[the remark this answers is not in the document]
Comment (gdesmott) [reply src/redisearch_rs/inverted_index/src/index_result.rs:180]: Let's keep this until `RSQueryTerm` is ported to Rust.

File: src/redisearch_rs/rqe_iterators/tests/inverted_index.rs [hunk 2]
```diff
@@ -126,18 +141,297 @@ impl<E: Encoder> BaseTest<E> {
...
+
+        // read the next element, docs_ids[4] has been removed so iterator should return the one after.
+        let doc = it
+            .read()
+            .expect("failed to read")
+            .expect("should not be at EOF");
+        assert_eq!(doc.doc_id, self.doc_ids[5]);
+        assert_eq!(it.last_doc_id(), self.doc_ids[5]);
+
+        // edge case: iterator is at the lost document which is then removed.
```
Comment (chesedo) [inline src/redisearch_rs/rqe_iterators/tests/inverted_index.rs]:
```suggestion
        // edge case: iterator is at the last document which is then removed.
```

File: src/redisearch_rs/rqe_iterators/tests/inverted_index.rs [hunk 3]
```diff
@@ -126,18 +141,297 @@ impl<E: Encoder> BaseTest<E> {
...
+    NumericTest::new(100).read();
 }
 
 #[test]
 /// test skipping from NumericFull iterator
 fn numeric_full_skip_to() {
-    let test = BaseTest::new(inverted_index::numeric::Numeric::default());
-    let mut it = NumericFull::new(test.ii.reader());
-    test.skip_to(&mut it);
+    NumericTest::new(100).skip_to();
```
Comment (chesedo) [inline src/redisearch_rs/rqe_iterators/tests/inverted_index.rs]: Btw, why are the tests this way? The content of `skip_to()` could just have been here right?

**Review by chesedo — COMMENTED** [review]
Looks good. Just have a question about the test structure

File: src/redisearch_rs/rqe_iterators/tests/inverted_index.rs (hunk 3 above)
Comment (gdesmott) [reply src/redisearch_rs/rqe_iterators/tests/inverted_index.rs]: Indeed, the extra method does not bring much.

I removed it in the previous PRs as it was introduced there.

**Review by chesedo — APPROVED** [review]
